### ISSUE TITLE
Add support for DNS SRV

### ISSUE THREAD

**Issue by pocman** [akka/akka-management#136]
I'm trying to use Akka Cluster Bootstrap in replacement for ConstructR-consul with akka-dns in my Lagom project.

Currently DNS SRV is not supported but I implemented the following to solve this issue : 
```scala
package akka.discovery.dns

import akka.actor.ActorSystem
import akka.discovery._
import akka.event.Logging
import akka.pattern.ask
import com.lightbend.dns.locator.{ServiceLocator => ServiceLocatorService}

import scala.concurrent.Future
import scala.concurrent.duration.FiniteDuration

/**
 * Looks for DNS SRV records for a given service.
 */
class DnsSrvServiceDiscovery(system: ActorSystem) extends SimpleServiceDiscovery {

  import SimpleServiceDiscovery._

  private val log = Logging(system, getClass)
  private val dns = system.actorOf(ServiceLocatorService.props, ServiceLocatorService.Name + "_DnsSrvServiceDiscovery")

  import system.dispatcher

  override def lookup(name: String, resolveTimeout: FiniteDuration): Future[Resolved] =
    dns
        .ask(ServiceLocatorService.GetAddresses(name))(resolveTimeout)
        .mapTo[ServiceLocatorService.Addresses]
        .map {
          case ServiceLocatorService.Addresses(addresses) =>
            log.info("Resolved Dns.Resolved: {}", addresses)
            addresses.map(sa => ResolvedTarget(sa.host, Some(sa.port))).to[collection.immutable.Seq]
        }.map(Resolved(name, _))
}
```

And modified the application.conf as follow :

```yaml
akka {
  discovery {
    method = akka-dns

    akka-dns {
      # currently no special configuration is present for this module
      class = akka.discovery.dns.DnsSrvServiceDiscovery
    }
  }
```

**Comment by ktoso** [comment]
Thanks! The issue about SRV is here: https://github.com/akka/akka-management/issues/72

We want to implement it natively, and then consolidate the other service locator API with this one, splitting out akka-discovery as it's own thing.

**Comment by pocman** [comment]
ServiceLocatorService is only resolving over UDP, this is a problem because [DNS through UDP has a 512 bytes limit](https://serverfault.com/questions/587625/why-dns-through-udp-has-a-512-bytes-limit).
I'm going to try to use https://github.com/spotify/dns-java

**Comment by pocman** [comment]
Here you go 
```
package akka.discovery.dns

import akka.actor.ActorSystem
import akka.discovery._
import akka.event.{Logging, LoggingAdapter}
import com.spotify.dns.{DnsSrvResolver, DnsSrvResolvers, LookupResult}
import com.typesafe.config.{Config, ConfigFactory}
import org.xbill.DNS.Lookup

import scala.collection.JavaConverters._
import scala.concurrent.Future
import scala.concurrent.duration.FiniteDuration

import com.spotify.dns.statistics.{DnsReporter, DnsTimingContext}

/**
 * Looks for DNS SRV records via tcp for a given service.
 */
class TcpDnsSrvServiceDiscovery(system: ActorSystem) extends SimpleServiceDiscovery {

  import SimpleServiceDiscovery._

  private val configuration: Config = ConfigFactory.load()
  System.setProperty("dns.server", configuration.getStringList("akka.io.dns.async-dns.nameservers").asScala.mkString(","))
  Lookup.getDefaultResolver.setTCP(true)

  private val log = Logging(system, getClass)
  private val reporter = new LoggingReporter(log)
  private val resolver: DnsSrvResolver = DnsSrvResolvers.newBuilder()
      .cachingLookups(false)
      .retainingDataOnFailures(true)
      .metered(reporter)
      .dnsLookupTimeoutMillis(1000)
      .build()

  import system.dispatcher

  override def lookup(name: String, resolveTimeout: FiniteDuration): Future[Resolved] = {
    Future {
      val lookups = resolver.resolve(name).asScala.toList
      println("Resolved Dns.Resolved: {}", lookups.mkString(", "))
      lookups.map {
        lookup: LookupResult => ResolvedTarget(lookup.host, Some(lookup.port))
      }
    }.map(Resolved(name, _))
  }

  private class LoggingReporter(log: LoggingAdapter) extends DnsReporter {
    override def resolveTimer: DnsTimingContext = new DnsTimingContext() {
      final private val start = System.currentTimeMillis

      override

      def stop(): Unit = {
        val now = System.currentTimeMillis
        val diff = now - start
        log.debug("Request took " + diff + "ms")
      }
    }

    override def reportFailure(error: Throwable): Unit = {
      log.error("Error when resolving: " + error)
    }

    override def reportEmpty(): Unit = {
      log.error("Empty response from server.")
    }
  }

}
```

**Comment by TimMoore** [comment]
Should this be closed as a duplicate of #72?

**Comment by chbatey** [comment]
Yep let's track this in #72
